src/main/java/com/conveyal/r5/analyst/GridComputer.java

```diff
@@ -0,0 +1,149 @@
+    private static final Logger LOG = LoggerFactory.getLogger(GridComputer.class);
+
+    /** SQS client. TODO: async? */
+    private static final AmazonSQS sqs = new AmazonSQSClient();
+
+    private static final Base64.Encoder base64 = Base64.getEncoder();
+
+    private final GridCache gridCache;
+
+    public final GridRequest request;
+
+    public final TransportNetwork network;
+
+    public GridComputer(GridRequest request, GridCache gridCache, TransportNetwork network) {
+        this.request = request;
+        this.gridCache = gridCache;
+        this.network = network;
+    }
+
+    public void run() throws IOException {
+        final Grid grid = gridCache.get(request.grid);
+
+        // ensure they all have the same zoom level
+        if (request.zoom != grid.zoom) throw new IllegalArgumentException("grid zooms do not match!");
+
+        // TODO cache these so they are not relinked? or is that fast enough it doesn't matter?
+        // NB this will skip all destinations that lie outside query bounds. This is intentional.
+        final WebMercatorGridPointSet targets =
+                new WebMercatorGridPointSet(request.zoom, request.west, request.north, request.width, request.height);
+
+        // use the middle of the grid cell
+        request.request.fromLat = Grid.pixelToLat(request.north + request.y + 0.5, request.zoom);
+        request.request.fromLon = Grid.pixelToLon(request.west + request.x + 0.5, request.zoom);
+
+        // Run the raptor algorithm to get times at each destination for each iteration
+        StreetMode mode;
+        if (request.request.accessModes.contains(LegMode.CAR)) mode = StreetMode.CAR;
+        else if (request.request.accessModes.contains(LegMode.BICYCLE)) mode = StreetMode.BICYCLE;
+        else mode = StreetMode.WALK;
+
+        LOG.info("Maximum number of rides: {}", request.request.maxRides);
+        LOG.info("Maximum trip duration: {}", request.request.maxTripDurationMinutes);
+
+        final LinkedPointSet linkedTargets = targets.link(network.streetLayer, mode);
+
+        StreetRouter sr = new StreetRouter(network.streetLayer);
+        sr.distanceLimitMeters = 2000;
+        sr.setOrigin(request.request.fromLat, request.request.fromLon);
+        sr.dominanceVariable = StreetRouter.State.RoutingVariable.DISTANCE_MILLIMETERS;
+        sr.route();
+
+        RaptorWorker router = new RaptorWorker(network.transitLayer, linkedTargets, request.request);
+
+        TIntIntMap reachedStops = sr.getReachedStops();
+
+        int millimetersPerSecond = (int) (request.request.walkSpeed * 1000);
+
+        for (TIntIntIterator it = reachedStops.iterator(); it.hasNext();) {
+            it.advance();
+            it.setValue(it.value() / millimetersPerSecond);
+        }
+
+        router.runRaptor(reachedStops, linkedTargets.eval(sr::getTravelTimeToVertex), new TaskStatistics());
+
+        int[] accessibilityPerIteration = new int[router.includeInAverages.cardinality()];
```

Comment:
Which begs the question of why we are calculating those bounds at all, we could make the algorithm faster if we didn't.